manifests/init.pp

```diff
@@ -86,101 +86,73 @@
-  $logrotate_compress    = 'nocompress',
-  $logrotate_files       = 4,
-  $logrotate_on          = false,
-  $logrotate_period      = 'daily',
-  $consul_ssl_enabled    = false,
-  $consul_ssl_verify     = true,
-  $consul_ssl_cert       = undef,
-  $consul_ssl_ca_cert    = undef,
-  $vault_enabled         = false,
-  $vault_address         = '',
-  $vault_token           = '',
-  $vault_ssl             = true,
-  $vault_ssl_verify      = true,
-  $vault_ssl_cert        = '',
-  $vault_ssl_ca_cert     = '',
-  $data_dir              = '',
-  $user                  = $::consul_template::params::user,
-  $group                 = $::consul_template::params::group,
-  $manage_user           = $::consul_template::params::manage_user,
-  $manage_group          = $::consul_template::params::manage_group,
-  $consul_reload_signal  = $::consul_template::params::reload_signal,
-  $consul_kill_signal    = $::consul_template::params::kill_signal,
-  $watches               = {},
+  Boolean $purge_config_dir                  = true,
+  $config_mode                               = $consul_template::params::config_mode,
+  $bin_dir                                   = '/usr/local/bin',
+  $arch                                      = $consul_template::params::arch,
+  $version                                   = $consul_template::params::version,
+  $install_method                            = $consul_template::params::install_method,
+  $os                                        = $consul_template::params::os,
+  $download_url                              = undef,
+  $download_url_base                         = $consul_template::params::download_url_base,
+  $download_extension                        = $consul_template::params::download_extension,
+  $package_name                              = $consul_template::params::package_name,
+  $package_ensure                            = $consul_template::params::package_ensure,
+  $config_dir                                = '/etc/consul-template',
+  $extra_options                             = '',
+  Boolean $service_enable                    = true,
+  Enum['stopped', 'running'] $service_ensure = 'running',
+  $consul_host                               = 'localhost',
+  Integer $consul_port                       = 8500,
+  $consul_token                              = '',
+  $consul_retry                              = '10s',
+  $consul_retry_attempts                     = '5',
+  $consul_retry_backoff                      = '250ms',
+  $consul_wait                               = undef,
+  $consul_max_stale                          = undef,
+  Boolean $deduplicate                       = false,
+  $deduplicate_prefix                        = undef,
+  $init_style                                = $consul_template::params::init_style,
+  $log_level                                 = $consul_template::params::log_level,
+  String $logrotate_compress                 = 'nocompress',
```

Comment:
this could be an enum, but leaving it as `String` for now to match the existing validate